### PR TITLE
chore(components/layout): description list strict mode support

### PR DESCRIPTION
**Pull request by Blackbaud-CoreyArcher**
[AB#2195421](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2195421)

### CODE REVIEW

File: libs/components/layout/src/lib/modules/description-list/description-list-term.component.ts [hunk 1]
```diff
@@ -20,5 +20,5 @@ export class SkyDescriptionListTermComponent {
     read: TemplateRef,
     static: true,
   })
-  public templateRef: TemplateRef<unknown>;
+  public templateRef!: TemplateRef<unknown> | undefined;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/layout/src/lib/modules/description-list/description-list-term.component.ts]: Forget to remove the `!`?

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/616/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/616/storybooks/ag-grid-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/616/storybooks/layout-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/616/storybooks/pages-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/616/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/616/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/616/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/616/playground/)

File: libs/components/layout/src/lib/modules/description-list/description-list-description.component.ts [hunk 2]
```diff
@@ -27,38 +27,43 @@ import { SkyDescriptionListService } from './description-list.service';
 export class SkyDescriptionListDescriptionComponent
   implements OnDestroy, OnInit
 {
-  public defaultDescription: string;
+  public defaultDescription: string | undefined;
 
-  public themeName: string;
+  public themeName = 'default';
```
Comment (Blackbaud-SteveBrush) [inline libs/components/layout/src/lib/modules/description-list/description-list-description.component.ts]: The theme's name is technically not defined unless the Theme service is provided. Can we keep this as `string | undefined`?

File: libs/components/layout/src/lib/modules/description-list/description-list-term.component.ts (hunk 1 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/layout/src/lib/modules/description-list/description-list-term.component.ts]: fixed

File: libs/components/layout/src/lib/modules/description-list/description-list-description.component.ts (hunk 2 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/layout/src/lib/modules/description-list/description-list-description.component.ts]: reverted

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**